### PR TITLE
[5779] extend rsync testtool to process parameters through json format files

### PR DESCRIPTION
**Pull request by konangc87**
|Related issue|
|---|
|[5779](https://github.com/wazuh/wazuh/issues/5779)|

## Description
This issue aims to extend the functionality of rsync testtool to accept parameters through json format files to exercise both dbsync and rsync libraries.

## DoD:
- [X] Implementation

### CODE REVIEW

File: src/shared_modules/rsync/testtool/oneTimeSync.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+{
+    rsync_close(m_rsyncHandle);
+}
+
+void OneTimeSync::syncData()
+{
+    const auto it{ m_inputData[0].find("dbsync") };
+    if (it != m_inputData[0].end())
+    {
+        // const auto dbsyncInput{ m_inputData[0].at("dbsync") };
```
Comment (tinachoB) [inline src/shared_modules/rsync/testtool/oneTimeSync.cpp:87]: Remove this please.

File: src/shared_modules/rsync/testtool/config.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
```
Comment (tinachoB) [inline src/shared_modules/rsync/testtool/config.json:1]: I think we should move this config.json and input.json files into a new "input" folder to separate implementation and input data files

File: src/shared_modules/rsync/testtool/oneTimeSync.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+#ifndef _ONE_TIME_SYNC_H_
+#define _ONE_TIME_SYNC_H_
+
+#include <thread>
+#include <chrono>
+#include <atomic>
```
Comment (tinachoB) [inline src/shared_modules/rsync/testtool/oneTimeSync.h:16]: Do we truly need these includes here?

**Review by tinachoB — CHANGES_REQUESTED** [review]

**Review by tinachoB — APPROVED** [review]
Looks great!

File: src/shared_modules/rsync/testtool/oneTimeSync.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ */
+#ifndef _ONE_TIME_SYNC_H_
+#define _ONE_TIME_SYNC_H_
+
+#include "json.hpp"
+#include "rsync.h"
+#include "dbsync.h"
+
+
+class OneTimeSync
```
Comment (Dwordcito) [inline src/shared_modules/rsync/testtool/oneTimeSync.h:19]: this could be "final" ?

File: src/shared_modules/rsync/testtool/oneTimeSync.cpp [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+                "Error in rsync_start_sync."
+            };
+        }
+    }
+}
+
+void OneTimeSync::rsyncCallback(const void* buffer, size_t bufferSize, void* userData)
+{
+    static unsigned int index{ 0 };
+    OneTimeSync* object{ reinterpret_cast<OneTimeSync*>(userData) };
```
Comment (Dwordcito) [inline src/shared_modules/rsync/testtool/oneTimeSync.cpp:144]: this could be const ?

File: src/shared_modules/rsync/testtool/oneTimeSync.cpp [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+}
+OneTimeSync::~OneTimeSync()
+{
+    rsync_close(m_rsyncHandle);
+}
+
+void OneTimeSync::syncData()
+{
+    const auto it{ m_inputData[0].find("dbsync") };
```
Comment (Dwordcito) [inline src/shared_modules/rsync/testtool/oneTimeSync.cpp:84]: use at()

File: src/shared_modules/rsync/testtool/oneTimeSync.cpp [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    if (it != m_inputData[0].end())
```
Comment (Dwordcito) [inline src/shared_modules/rsync/testtool/oneTimeSync.cpp:85]: use at()

File: src/shared_modules/rsync/testtool/oneTimeSync.cpp [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+            {
+                "Error in dbsync_sync_row."
+            };
+        }
+    }
+}
+
+void OneTimeSync::pushData()
+{
+    const auto it{ m_inputData[0].find("rsync_push") };
```
Comment (Dwordcito) [inline src/shared_modules/rsync/testtool/oneTimeSync.cpp:102]: use at()

File: src/shared_modules/rsync/testtool/oneTimeSync.cpp [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+    if (it != m_inputData[0].end())
```
Comment (Dwordcito) [inline src/shared_modules/rsync/testtool/oneTimeSync.cpp:103]: use at()

File: src/shared_modules/rsync/testtool/oneTimeSync.cpp [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+                    "Error in rsync_push_message."
+                };
+            }
+        }
+    }
+}
+
+void OneTimeSync::startSync()
+{
+    const auto it{ m_inputData[0].find("rsync_start_sync") };
```
Comment (Dwordcito) [inline src/shared_modules/rsync/testtool/oneTimeSync.cpp:123]: use at()

File: src/shared_modules/rsync/testtool/cmdArgsHelper.h [hunk 11]
```diff
@@ -21,19 +21,61 @@ class CmdLineArgs
 {
 public:
     CmdLineArgs(const int argc, const char* argv[])
-    : m_outputFolder{ paramValueOf(argc, argv, "-o") }
-    , m_updatePeriod{ std::stoul(paramValueOf(argc, argv, "-u")) }
-    {}
+    : m_updatePeriod{ 0 }
+    {
```
Comment (Dwordcito) [inline src/shared_modules/rsync/testtool/cmdArgsHelper.h:25]: this class could be final ?

File: src/shared_modules/rsync/testtool/oneTimeSync.cpp [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+#include "oneTimeSync.h"
+#include <fstream>
+#include <sstream>
+
```
Comment (Dwordcito) [inline src/shared_modules/rsync/testtool/oneTimeSync.cpp:14]: remove \n

**Review by tinachoB — APPROVED** [review]

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**